server/src/github.com/alexeyknyshev/cpsrv/cpsrv_patch_test.go

```diff
@@ -0,0 +1,425 @@
+	TaranResp, err := hCtx.Tnt().Call("cashpointProposePatch", []interface{}{requestJson})
+	if err != nil {
+		t.Errorf("Tnt cashpointProposePatch call err: %v", err)
+	}
+	CpId := TaranResp.Data[0].([]interface{})[0]
+	if CpId.(uint64) == uint64(0) {
+		t.Error("Failed to create patch, CpId == 0")
+		return 0, 0
+	} else {
+		fmt.Println("\nCreate new patch, CpId = ", CpId)
+	}
+
+	resp, err := hCtx.Tnt().Call("getCashpointPatches", []interface{}{CpId})
+	if err != nil {
+		t.Errorf("Tnt getCashpointPatches call err: %v", err)
+	}
+	byteResp := []byte(resp.Data[0].([]interface{})[0].(string))
+	lastPatch := searchLastPatch(t, byteResp)
+	return lastPatch, CpId.(uint64)
+}
+
+func TestPatchCreateNewCP(t *testing.T) {
+
+	patchReq := getPatchExampleNewCP()
+	request := TestPatchReq{
+		Data:   *patchReq,
+		UserId: 1,
+	}
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		t.Fatalf("Json Marshal error %v", err)
+	}
+	fmt.Println("Request:\n", string(requestJson), "\n")
+
+	hCtx, err := makeHandlerContext(getServerConfig())
+	if err != nil {
+		t.Fatalf("Connection to tarantool failed: %v", err)
+	}
+
+	defer hCtx.Close()
+
+	metrics, err := getSpaceMetrics(hCtx)
+	if err != nil {
+		t.Errorf("Failed to get space metric on start: %v", err)
+	}
+	defer checkSpaceMetrics(t, func() ([]byte, error) { return getSpaceMetrics(hCtx) }, metrics)
+
+	lastPatch, CpId := invokeTaranPatchFuncs(t, hCtx, requestJson)
+	if lastPatch == 0 {
+		return
+	}
+	resp, err := hCtx.Tnt().Call("getCashpointPatchByPatchId", []interface{}{lastPatch})
+	resPatch := resp.Data[0].([]interface{})
+	expPatchData := "{\"id\":" + strconv.FormatInt(int64(CpId), 10) + "}"
+	expectedPatch := []interface{}{uint64(lastPatch), CpId, uint64(request.UserId), expPatchData, uint64(0)}
+	comparePatches(t, resPatch, expectedPatch)
+	fmt.Println("Delete cashpoint ", CpId)
+	resp, err = hCtx.Tnt().Call("deleteCashpointById", []interface{}{CpId})
+	if err != nil {
+		t.Errorf("Tnt call deleteCashpointById err: %v", err)
+	}
+	if !resp.Data[0].([]interface{})[0].(bool) {
+		t.Error("deleteCashpointById return false")
+	}
+}
+
+func TestPatchChangeExistCP(t *testing.T) {
+
+	patchReq, expPatchData := getPatchExampleExistCP()
+	request := TestPatchReq{
+		Data:   *patchReq,
+		UserId: 1,
+	}
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		t.Fatalf("Json Marshal error %v", err)
+	}
+	fmt.Println("Request:\n", string(requestJson), "\n")
+
+	hCtx, err := makeHandlerContext(getServerConfig())
+	if err != nil {
+		t.Fatalf("Connection to tarantool failed: %v", err)
+	}
+	defer hCtx.Close()
+
+	metrics, err := getSpaceMetrics(hCtx)
+	if err != nil {
+		t.Errorf("Failed to get space metric on start: %v", err)
+	}
+	defer checkSpaceMetrics(t, func() ([]byte, error) { return getSpaceMetrics(hCtx) }, metrics)
+
+	lastPatch, CpId := invokeTaranPatchFuncs(t, hCtx, requestJson)
+	if lastPatch == 0 {
+		return
```

Comment:
Yes, invokeTaranPatchFuncs contain Error call. Thx for notice, I have seen suspicious code in that function